packages/web-frontend/src/sagas.js

```diff
@@ -1048,8 +1048,8 @@ function* watchLogoutSuccess() {
 function* fetchLoginData(action) {
   if (action.loginType === LOGIN_TYPES.MANUAL) {
     const { routing: location } = yield select();
-    yield put(setOverlayComponent(null));
     yield call(fetchProjectData);
+    yield put(setOverlayComponent(LANDING));
```

Comment:
```suggestion
    yield call(fetchProjectData);
    const { PROJECT } = decodeLocation(location);
    const overlay = PROJECT === 'explore' ? LANDING : null;
    yield put(setOverlayComponent(overlay));```

I think that should do it.
ps. I assume that it is just on the 'explore' project that we want to show the landing page after login.

Reply:
I think that should do it.
ps. I assume that it is just on the 'explore' project that we want to show the landing page after login.

Reply:
Hrm...this seems to work, if I go into private project url unlogged in, then it brings up log in form, after log in sucess it brings up the project page briefly then closes it. This may be acceptable?

Reply:
I guess the other senerio that it should bring up the project page is where a user links into a project they don't have access to.

Reply:
Actually no, if a user doesn't have access to the project after logging in, they get the request access modal, which is ideal!
Only the half a second flash of the project page before closing seems to be the only issue.

Reply:
hmmm given that this is a hotfix, I'd say it is acceptable to have the flash. Obviously it's not ideal but it is also not the main flow that users take. I had a quick look and am not sure why there is the flash as the default state for the overlay is null which was my first thought. Maybe have a quick look at fixing it and if you can't nail it down in then create a bug card for it to be picked up later.